src/box/index.h

```diff
@@ -689,6 +718,10 @@ index_replace(struct index *index, struct tuple *old_tuple,
 	      struct tuple *new_tuple, enum dup_replace_mode mode,
 	      struct tuple **result, struct tuple **successor)
 {
+	if (!box_check_deadline()) {
+		diag_set(ClientError, ER_SPACE_ITERATION_TIMEOUT);
+		return -1;
+	}
```

Comment:
There's no point in checking the timeout in DML statements, because they usually yield anyway - on WAL write. As for other methods (get/min/max/random/size) - they are O(1) so they don't need the check, either.

Reply:
1. `get` is not O(1), and we wanted to check deadline in it.
2. We decided to check DML.